src/zero_copy/text.rs

```diff
@@ -88,3 +130,368 @@ where
 
     go_extra!();
 }
+
+/// A parser that accepts (and ignores) any number of whitespace characters.
+///
+/// The output type of this parser is `()`.
+#[must_use]
+pub fn whitespace<I, E, S>() -> impl for<'a> Parser<'a, I, E, S, Output = ()>
+where
+    I: SliceInput + ?Sized,
+    I::Token: Char,
+    E: Error<I>,
+    for<'a> S: 'a,
+{
+    primitive::any()
+        .filter(|x: &I::Token| x.is_whitespace())
+        .repeated()
+        .ignored()
+}
+
+/// A parser that accepts (and ignores) any newline characters or character sequences.
+///
+/// The output type of this parser is `()`.
+///
+/// This parser is quite extensive, recognising:
+///
+/// - Line feed (`\n`)
+/// - Carriage return (`\r`)
+/// - Carriage return + line feed (`\r\n`)
+/// - Vertical tab (`\x0B`)
+/// - Form feed (`\x0C`)
+/// - Next line (`\u{0085}`)
+/// - Line separator (`\u{2028}`)
+/// - Paragraph separator (`\u{2029}`)
+#[must_use]
+pub fn newline<S, E>() -> impl for<'a> Parser<'a, str, E, S, Output = ()>
+where
+    E: Error<str>,
+    for<'a> S: 'a,
+{
+    just('\r')
+        .or_not()
+        .ignore_then(just('\n'))
+        .or(just('\r')) // Carriage return
+        .or(just('\x0B')) // Vertical tab
+        .or(just('\x0C')) // Form feed
+        .or(just('\u{0085}')) // Next line
+        .or(just('\u{2028}')) // Line separator
+        .or(just('\u{2029}')) // Paragraph separator
+        .ignored()
+}
+
+/// A parser that accepts one or more ASCII digits.
+///
+/// The `radix` parameter functions identically to [`char::is_digit`]. If in doubt, choose `10`.
+#[must_use]
+pub fn digits<'a, I, E, S, C>(
+    radix: u32,
+) -> Map<
+    RepeatedSlice<impl Parser<'a, I, E, S, Output = I::Token>, I, E, S>,
+    fn(&'a <I as SliceInput>::Slice) -> C,
+>
+where
+    I: SliceInput + ?Sized,
+    I::Token: Char,
+    <I as SliceInput>::Slice: 'a,
+    E: Error<I>,
+    S: 'a,
+    C: From<&'a <I as SliceInput>::Slice>,
+{
+    primitive::any()
+        .filter(move |x: &I::Token| x.is_digit(radix))
+        .repeated_slice()
+        .at_least(1)
+        .map(C::from)
```

Comment:
I'm not sure I understand what this `C::from` does. How is the conversion occurring, and into what?